velox/functions/sparksql/tests/ArithmeticTest.cpp

```diff
@@ -401,5 +401,94 @@ TEST_F(LogNTest, log10) {
   EXPECT_EQ(log10(kInf), kInf);
 }
 
+class RandTest : public SparkFunctionBaseTest {
+ public:
+  RandTest() {
+    // This setting is needed as we expect int32 type is parsed for
+    // partitionIndex literal.
+    options_.parseIntegerAsBigint = false;
+  }
+
+ protected:
+  std::optional<double> rand(int32_t seed, int32_t partitionIndex = 0) {
+    return evaluateOnce<double>(
+        fmt::format("rand({}, {})", seed, partitionIndex),
+        makeRowVector(ROW({}), 1));
+  }
+
+  std::optional<double> randWithNullSeed(int32_t partitionIndex = 0) {
+    return evaluateOnce<double>(
+        fmt::format("rand(NULL, {})", partitionIndex),
+        makeRowVector(ROW({}), 1));
+  }
+
+  std::optional<double> randWithNoSeed() {
+    return evaluateOnce<double>("rand()", makeRowVector(ROW({}), 1));
+  }
+
+  VectorPtr randWithBatchInput(int32_t seed, int32_t partitionIndex = 0) {
+    auto exprSet = compileExpression(
+        fmt::format("rand({}, {})", seed, partitionIndex), ROW({}));
+    return evaluate(*exprSet, makeRowVector(ROW({}), 20));
+  }
+
+  void checkResult(const std::optional<double>& result) {
+    EXPECT_NE(result, std::nullopt);
+    EXPECT_GE(*result, 0.0);
+    EXPECT_LT(*result, 1.0);
+  }
+
+  bool AreSameVectors(const VectorPtr& vector1, const VectorPtr& vector2) {
```

Comment:
Use assertEqualVectors from velox/vector/tests/utils/VectorTestBase.h

Reply:
I intended to test they are different for each partition. Just kept this function and changed the function name to make it clear. Thanks!